Libs/Goals/AdhocGoal.cs

```diff
@@ -47,13 +47,13 @@ public override async Task PerformAction()
             {
                 await this.stopMoving.Stop();
 
-                if (playerReader.PlayerBitValues.IsMounted)
-                {
-                    await wowProcess.Dismount();
-                }
                 await Task.Delay(1000);
             }
-
+            if (playerReader.PlayerBitValues.IsMounted)
```

Comment:
Use  StopBeforeCast property for this!

```
if (key.StopBeforeCast)
            {
                await this.stopMoving.Stop();

                if (playerReader.PlayerBitValues.IsMounted)
                {
                    await wowProcess.Dismount();
                }
                await Task.Delay(1000);
            }
```

Reply:
Can you undo the changes to this file and I can then accept the pull.

Reply:
Done .